### PR TITLE
Don/docs

### PR DESCRIPTION
**Pull request by dzereski**
This is the first pass of the Sphinx-generated docs for the Indico Python Client.  This pass has the basic Sphinx framework with a fairly comprehensive User Guide and examples.  Docstrings are currently being pulled from client.client.py, config.config.py and queries.documents.py.  Want to get this bit of work merged then finish the rest by EOD on Friday.

### CODE REVIEW

File: examples/dir-extract.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+from indico import IndicoClient
+from indico.config import IndicoConfig
+from indico.queries.documents import DocumentExtraction
+
+
+def ocr_files(target_files):
+
+    my_config = IndicoConfig(
+        host='dev.indico.io',
+        api_token_path=Path(__file__).parent / 'indico_api_token.txt'
```
Comment (nkassis) [inline examples/dir-extract.py]: For the example I think we assume it's in their home dir?

File: examples/dir-extract.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+from indico import IndicoClient
+from indico.config import IndicoConfig
+from indico.queries.documents import DocumentExtraction
+
+
+def ocr_files(target_files):
+
+    my_config = IndicoConfig(
+        host='dev.indico.io',
```
Comment (nkassis) [inline examples/dir-extract.py]: Definitively make this app.indico.io

File: examples/dir-extract.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    failed = []
+
+    for src_path in target_files:
+        finfo = src_path.stat()
+
+        if finfo.st_size > 0:
+            start_time = time.time()
+            job = client.call(DocumentExtraction(files=[src_path], json_config='{"preset": "simple"}'))
+            upload_time = time.time() - start_time
+            if job is not None:
```
Comment (nkassis) [inline examples/dir-extract.py]: this is a list of jobs so I do if jobs.

File: examples/dir-extract.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        host='dev.indico.io',
+        api_token_path=Path(__file__).parent / 'indico_api_token.txt'
+    )
+
+    client = IndicoClient(config=my_config)
+
+    jobs = []
+    failed = []
+
+    for src_path in target_files:
```
Comment (nkassis) [inline examples/dir-extract.py]: Doc extraction takes a list of files

File: examples/dir-extract.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        else:
+            print(f"ERROR - Zero length file {src_path.parent.name}/{src_path.name} - skipping")
+            failed.append(src_path)
+    
+    total_jobs = len(jobs)
+    n_finished = 0
+    print(f"\nQueued {total_jobs} total files for OCR")
+    print("=" * 40)
+
+    for path, jobs in zip(target_files, jobs):
```
Comment (nkassis) [inline examples/dir-extract.py]: this has to be rewriten with the new JobsStatus query

File: indico/client/client.py [hunk 6]
```diff
@@ -22,8 +39,18 @@ def _handle_request_chain(self, chain: RequestChain):
 
     def call(self, request: Union[HTTPRequest, RequestChain]):
         """
-        Make a call to the indico IPA platform
+        Make a call to the Indico IPA Platform
+
+        Args:
+            request (HTTPRequest or RequestChain): GraphQL request to send to the Indico Platform
```
Comment (nkassis) [inline indico/client/client.py]: Graphql or HTTPRequest to make to the platform

File: indico/client/client.py [hunk 7]
```diff
@@ -22,8 +39,18 @@ def _handle_request_chain(self, chain: RequestChain):
...
+        
+        Returns:
+            JSON Response of the request
```
Comment (nkassis) [inline indico/client/client.py]: Not always true, return T depending of the type returned by the request class

File: examples/dir-extract.py (hunk 1 above)
Comment (dzereski) [reply examples/dir-extract.py]: Yeah, I'm going to hold off on this example until wait on any complete job is done.

File: examples/dir-extract.py (hunk 3 above)
Comment (dzereski) [reply examples/dir-extract.py]: Example removed

File: examples/dir-extract.py (hunk 4 above)
Comment (dzereski) [reply examples/dir-extract.py]: Example removed

File: examples/dir-extract.py (hunk 2 above)
Comment (dzereski) [reply examples/dir-extract.py]: Example removed

File: examples/dir-extract.py (hunk 5 above)
Comment (dzereski) [reply examples/dir-extract.py]: Example removed

File: indico/client/client.py (hunk 6 above)
Comment (dzereski) [reply indico/client/client.py]: Changed HTTPRequest to GraphQLRequest

File: indico/client/client.py (hunk 7 above)
Comment (dzereski) [reply indico/client/client.py]: Changed to...
Returns:
    Response appropriate to the class of the provided request parameter. Often JSON but not 
    always.

**Pull request merged**